src/Plugin/Field/FieldWidget/AutomatedListingConfigurationWidgetEnhanced.php

```diff
@@ -0,0 +1,814 @@
+          }
+        }
+      }
+    }
+
+    // Today filter.
+    $date_fields = $this->indexHelper->getIndexDateFields($this->index);
+    $element['today'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Filters for content types that include dates'),
+      '#description' => $this->t('You can apply more filters to Event, Grant, Family violence recommendations and Publications content, if a Start date and End date has been included.'),
+      '#open' => TRUE,
+      '#collapsible' => TRUE,
+      '#group_name' => 'filters_today',
+      '#states' => [
+        'invisible' => [
+          ':input[name="' . $this->getFormStatesElementName('type', $items, $delta, $element) . '"]' => ['value' => '']
+        ]
+      ],
+    ];
+
+    $element['today']['status'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Enable date filter'),
+      '#default_value' => $configuration['filter_today']['status'] ?? FALSE,
+    ];
+    $default_filter_today_start_date = $configuration['filter_today']['start_date'] ?? 'field_event_date_start_value';
+    if (!isset($date_fields[$default_filter_today_start_date])) {
+      $default_filter_today_start_date = '';
+    }
+    $default_filter_today_end_date = $configuration['filter_today']['end_date'] ?? 'field_event_date_end_value';
+    if (!isset($date_fields[$default_filter_today_end_date])) {
+      $default_filter_today_end_date = '';
+    }
+    if (isset($configuration['filter_today']['status']) && empty($configuration['filter_today']['status'])) {
+      $element['today']['#open'] = FALSE;
+    }
+
+    $element['today']['start_date'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Start Date'),
+      '#default_value' => $default_filter_today_start_date,
+      '#options' => ['' => $this->t('- No mapping -')] + $date_fields,
+    ];
+
+    $element['today']['end_date'] = [
+      '#type' => 'select',
+      '#title' => $this->t('End Date'),
+      '#default_value' => $default_filter_today_end_date,
+      '#options' => ['' => $this->t('- No mapping -')] + $date_fields,
+    ];
+    $element['today']['criteria'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Select the date filter for your collection'),
+      '#default_value' => $configuration['filter_today']['criteria'] ?? 'upcoming',
+      '#options' => [
+        'all' => $this->t('All dates'),
+        'upcoming' => $this->t('Upcoming dates'),
+        'from_current' => $this->t('Upcoming and current dates'),
+        'past' => $this->t('Past dates'),
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function massageFormValues(array $values, array $form, FormStateInterface $form_state) {
+    $values = parent::massageFormValues($values, $form, $form_state);
+    foreach ($values as $delta => &$value) {
+      $config = [];
+      $config['index'] = $this->fieldDefinition->getFieldStorageDefinition()->getSetting('index');
+
+      /** Deprecated Fields Start */
```

Comment:
CS issue: C-style inline comment is discouraged within functions, and `/** */` should be only used to start a PHPDoc block.